lib/core/src/Cardano/Wallet/Primitive/AddressDiscovery/Random.hs

```diff
@@ -49,6 +49,9 @@ import GHC.Generics
 newtype RndState = RndState { getRndState :: RndKey 'RootK XPrv }
     deriving (Generic)
 
+instance Show RndState where
+    show _ = "RndState <xprv>"
```

Comment:
What's the rationale for this custom `Show` instance :thinking: ? 
Writing custom `Show` is always painful, because there's in theory parentheses to handle correctly...

Do we need this instance? I get that we can't derive it automatically because there's no `Show` instance on the `XPrv`, so I am just wondering.

Reply:
yes, it seems to clash with the instance already present in https://github.com/input-output-hk/cardano-wallet/blob/master/lib/core/test/unit/Cardano/Wallet/Primitive/AddressDiscovery/RandomSpec.hs#L265 -> overlapping has to be introduced in `RandomSpec`

Reply:
`Show` instance seems not to be needed anymore. Removed, rebased the branch.